src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -508,6 +510,49 @@ public BoundStatementList StatementList(BoundStatement first, BoundStatement sec
             return new BoundStatementList(Syntax, ImmutableArray.Create(first, second)) { WasCompilerGenerated = true };
         }
 
+        [return: NotNullIfNotNull(nameof(first)), NotNullIfNotNull(nameof(second))]
+        public BoundStatement? Concat(BoundStatement? first, BoundStatement? second)
+            => (first == null) ? second : (second == null) ? first : StatementList(first, second);
+
+        public BoundBlockInstrumentation Instrumentation(BoundBlockInstrumentation? innerInstrumentation = null, LocalSymbol? local = null, BoundStatement? prologue = null, BoundStatement? epilogue = null)
```

Comment:
"CreateOrAppendInstrumentation"?

Reply:
This creates `BoundBlockInstrumentation` instance, similar to the `Block` method above. Would `BlockInstrumentation` be better?

Reply:
I think this method is not like other methods. It doesn't simply incorporate all inputs, therefore, I think it would be good to reflect the fact in the name. This is a form of a merge, rather than a plain creation.

Reply:
To answer the question, I think "BlockInstrumentation" doesn't reflect the nature of the method

Reply:
`CombineInstrumentation`?